keras/losses.py

```diff
@@ -1496,6 +1500,10 @@ def mean_absolute_percentage_error(y_true, y_pred):
   diff = tf.abs(
       (y_true - y_pred) / backend.maximum(tf.abs(y_true),
                                           backend.epsilon()))
+  if ignore_zeros is True:
+    mask = y_true != 0
```

Comment:
You might want to add some test for different dtypes (floats), I would expect `y_true != 0` might raise some error when dtype doesn't match.